### DISCUSSION TITLE
Optimizing TMC Parameters

### DISCUSSION THREAD

**Discussion by Sineos** [zapta/ble_stepper_motor_analyzer#7] [General]
I have been playing around with the analyzer and Klipper. I was using different TMC settings like `driver_TBL`, `driver_TOFF`, `driver_HEND` and `driver_HSTRT` and the information in https://www.trinamic.com/fileadmin/assets/Support/AppNotes/AN001-SpreadCycle.pdf
Whereas these parameters are suitable to make the stepper sound like a dying alien, I could not see any striking differences in the visual representation of the analyzer app. 

Did anybody have more success or any pointer to share? I have been using my cheap oscilloscope to this end already but was hoping for a more straight forward opportunity.

**Comment by zapta** [comment]
When you use your oscilloscope, did you measure current or voltage?  It's
possible that the voltage has different patterns or frequencies but they
result in a similar current pattern because of the large ratio between the
voltage (chopper) frequency and the stepper's cycle frequency.  Having the
noisy JLCPCB sensors doesn't help either to see tiny current patterns.

Zapta.

**Reply by Sineos** [reply]
I was measuring current and voltage but due to the fact that my oscilloscope is only 2 channels, it was always a challenge. TBH, I did this once and found it much too involved for the results I was able to obtain.
This does not mean that:
1. One could achieve better results than I did
2. Do it easier

Basically the reason, why I have some hopes in the analyzer and started this discussion,

**Comment by zapta** [comment]
Edie the Engineer from the Voron team recently had a video that you may
find interesting.

https://www.youtube.com/watch?v=RWDZLtUeg6o&t=995s

**Reply by Sineos** [reply]
Thanks. Good explanation, but basically what is in the application note of Trinamic. I was searching for a less complex way :-)

**Comment by Sineos** [comment]
Do you think this cool piece of hard- / software could be extended to do the process as described in the above linked Trinamic Application Note?
As these stepper drivers are incredibly popular, I think this would be a real game changer to have easy access to such a possibility.
